src/component/SideBar.tsx

```diff
@@ -0,0 +1,94 @@
+import styled from '@emotion/styled'
+
+
+const Xbutton = styled.button`
+    color: #fff;
+    font-size: 40px;
+`
+function Closebutton(){
+    return <Xbutton>×</Xbutton>;
+}
+
+const TaskBottomButton = styled.div`
+    display: flex;
+    justify-content: space-between;
+    width: 140px;
+`
+const TaskBottomButtonChild  = styled.div`
+    background: #BF5745;
+    width: 3px;
+    height: 10px;
+    
+`
+
+const TextButton = styled.button`
+    margin-top: 20px;
+    color: #fff;
+    background: #BF5745;
+    padding: 10px 30px;
+`
+
+function TaskButton(){
+    return (
+        <div>
+            <TextButton>タスク管理</TextButton>
+            <TaskBottomButton>
+                <TaskBottomButtonChild />
+                <TaskBottomButtonChild />
+            </TaskBottomButton>
+        </div>
+    )
+}
+function AddGroupButton(props: { isLogin: boolean; }){
+    const text = props.isLogin ? "＋ グループ追加" : "ログイン";
+    return (
+        <TextButton>{text}</TextButton>
+    )
+}
+const Line = styled.hr`
+    background-color: #fff;
+    height: 2px;
+    margin-top: 20px;
+`
+const GroupButton = styled.button`
+    background: #fff;
+    border: 3px solid #BF5745;
+    color: #1B2668;
+    border-radius: 10px;
+    padding: 10px 30px;
+    margin-top: 5px;
+`
+function GroupList(props: { groupList: string[]; }){
+    const items=[];
+    for (let i=0; i<props.groupList.length;i++){
+        items.push(<div><GroupButton>{props.groupList[i]}</GroupButton></div>)
+    }
+    return <div>{items}</div>;
```

Comment:
ここの部分高階関数を使って
```suggestion
    return (
      <div>
        {props.groupList.map(group => (<div key={group}><GroupButton>{group}</GroupButton></div>))}
      </div>
    )
```
みたいに書いてほしいです :pray:
ref: https://react.dev/learn/rendering-lists